mlflow/pytorch/torchserve/tests/test_plugin.py

```diff
@@ -0,0 +1,217 @@
+import atexit
+import json
+import os
+import pytest
+import shutil
+import subprocess
+import time
+from mlflow import deployments
+from mlflow.exceptions import MlflowException
+
+f_target = "torchserve"
+f_deployment_id = "test"
+f_flavor = None
+f_model_uri = "mlflow/pytorch/torchserve/tests/resources/linear.pt"
+
+env_version = "1.0"
+env_model_file = "mlflow/pytorch/torchserve/tests/resources/linear_model.py"
+env_handler_file = "mlflow/pytorch/torchserve/tests/resources/linear_handler.py"
+sample_input_file = "mlflow/pytorch/torchserve/tests/resources/sample.json"
+sample_output_file = "mlflow/pytorch/torchserve/tests/resources/output.json"
+
+
+@pytest.fixture(scope="session")
+def start_torchserve():
+    if not os.path.isdir("model_store"):
+        os.makedirs("model_store")
+    cmd = "torchserve --start --model-store {}".format("./model_store")
+    return_code = os.system(cmd)
+
+    count = 0
+    for _ in range(5):
+        value = health_checkup()
+        if (
+            value is not None
+            and value != ""
+            and json.loads(value)["status"] == "Healthy"
+        ):
+            time.sleep(1)
+            break
+        else:
+            count += 1
+            time.sleep(5)
+    if count >= 5:
+        raise Exception("Unable to connect to torchserve")
+    return True
+
+
+def health_checkup():
+    curl_cmd = "curl http://localhost:8080/ping"
+    (value, err) = subprocess.Popen(
+        [curl_cmd], stdout=subprocess.PIPE, shell=True
+    ).communicate()
+    return value.decode("utf-8")
+
+
+def stop_torchserve():
+    cmd = "torchserve --stop"
+    return_code = os.system(cmd)
+
+    if os.path.isdir("model_store"):
+        shutil.rmtree("model_store")
+
+
+@pytest.fixture
+def clear_env_variables():
+    if "VERSION" in os.environ:
+        os.environ.pop("VERSION")
+    if "MODEL_FILE" in os.environ:
+        os.environ.pop("MODEL_FILE")
+    if "HANDLER_FILE" in os.environ:
+        os.environ.pop("HANDLER_FILE")
```

Comment:
Nit: you should be able to use Python's `mock` library to declaratively mock environment variables in tests, rather than imperatively clearing them at the start of each test via this fixture, e.g:

```
with mock.patch.dict(os.environ, {"VERSION": "my-custom-version"}):
  # run test code
```